### PR TITLE
feat: check all autosquash messages

### PR DESCRIPTION
**Pull request by JP-Ellis**
Expand the autosquash prefixes from just `fixup!` to now include `squash!` and `amend!`. These are the prefixes supported by default when using [Git's `--autosquash`](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---autosquash).

To keep the commits atomic as requested in #394, this  commit _only_ changes the functionality, while leaving the configuration as is (thereby creating an mismatch between the config naming, and resulting behaviour)

EDIT: New commit message will be:

Committed, until now, only supported the `fixup!` prefix. The standard Git implementation of `--autosquash` also supports `squash!` and `amend!`. This commit modifies the logic of the handling of `fixup!` to support the other two prefixes in the same way.

Note that this purely changes the logic. Function names and options remain unchanged to ensure backwards compatibility. It may be worth considering renaming this to `autosquash` given it is motivated primarily by Git's `--autosquash` flag.

Ref: https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---autosquash

### CODE REVIEW

**Comment by epage** [comment]
> `feat!`

For myself, I don't think I'd consider this a major breaking change.  Its a minor breaking change that expands functionality to cover cases that either didn't exist or were unknown at the time of creation.

**Comment by JP-Ellis** [comment]
I would have considered this to be a breaking change, but if you don't, you should then use a squash-and-rebase instead of a merge. This way, you can ensure the commit message is exactly as you want.

**Comment by epage** [comment]
This is blocked on CI.  Since you have to edit this anyways, please remove `!`.

**Comment by JP-Ellis** [comment]
As requested I have removed the breaking change exclamation from the commit message. I do not know if CI is fixed yet though.

File: crates/committed/src/report.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ pub(crate) enum Content<'s> {
     NoPunctuation(NoPunctuation),
     Imperative(Imperative<'s>),
     Wip(Wip),
-    Fixup(Fixup),
+    Autosquash(Autosquash),
```
Comment (epage) [inline crates/committed/src/report.rs:66]: This change is unrelated to this commit and *is* a breaking change as discussed on the other PRs

Comment (epage) [reply crates/committed/src/report.rs:66]: To be clear, my expectation is this commit will continue to emit `Fixup`

**Comment by epage** [comment]
Looks like `cargo fmt` also needs to be run.

**Comment by JP-Ellis** [comment]
Rebased the PR on top of master so it can merge.
